test/e2e/framework/pod/resource.go

```diff
@@ -298,6 +298,26 @@ func podsRunning(c clientset.Interface, pods *v1.PodList) []error {
 	return e
 }
 
+func podContainerFailed(c clientset.Interface, namespace, podName string, containerIndex int, reason string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pod, err := c.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		switch pod.Status.Phase {
+		case v1.PodPending:
+			containerStatus := pod.Status.ContainerStatuses[containerIndex]
```

Comment:
This part makes an error like
```
test/e2e/framework/framework.go:597
Test Panicked
staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55

Panic: runtime error: index out of range [0] with length 0

Full stack:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x486a820, 0xc003066c00)
	GOROOT/src/runtime/panic.go:969 +0x175
k8s.io/kubernetes/test/e2e/framework/pod.podContainerFailed.func1(0xc00299b360, 0x7f6fc02cb7f0, 0xc001c75440)
	test/e2e/framework/pod/resource.go:309 +0x3c6
```

Reply:
`pod.Status.ContainerStatueses` can be length 0.
So it is better to add the following check before current line 309
```
if len(status.ContainerStatuses) == 0 {
    return false, nil
}
```

Reply:
good point, I've included this.